File: WeeklyChallenge/2주차_상호평가/solution.go
package solution

func minMaxFromArray(list []int) (int, int) {
	var smallest, biggest int
	smallest, biggest = list[0], list[0]
	for _, v := range list {
		if v > biggest {
			biggest = v
		}
		if v < smallest {
			smallest = v
		}
	}
	return smallest, biggest
}

func getAvg(list []int) int {
	var sum int
	sum = 0
	for _, v := range list {
		sum += v
	}

	return sum / len(list)
}

func solution(scores [][]int) string {
	m := map[int]rune{
		10: 'A',
		9:  'A',
		8:  'B',
		7:  'C',
		6:  'D',
		5:  'D',
		4:  'F',
		3:  'F',
		2:  'F',
		1:  'F',
		0:  'F',
	}

	var answer []int

	for i := 0; i < len(scores); i++ {
		myScore := scores[i][i]
		temp := []int{}
		for j := 0; j < len(scores); j++ {
			if j == i {
				continue
			}
			temp = append(temp, scores[j][i])
		}
		minimum, maximum := minMaxFromArray(temp)
		if myScore >= minimum && myScore <= maximum {
			temp = append(temp, myScore)
		}
		answer = append(answer, getAvg(temp))
	}

	result := ""
	for _, v := range answer {
		result += string(m[v/10])
	}
	return result
}

File: WeeklyChallenge/2주차_상호평가/solution.py
def getScore(score):
    n = score//10
    if n >= 9:
        return 'A'
    elif n == 8:
        return 'B'
    elif n == 7:
        return 'C'
    elif n >= 5:
        return 'D'
    return 'F'


def solution(scores):
    avg = []
    
    for stdNum in range(len(scores)):
        myScore = scores[stdNum][stdNum]
        temp = []
        
        for i in range(len(scores)):
            if i != stdNum:
                temp.append(scores[i][stdNum])
        if myScore > max(temp) or myScore < min(temp):
            avg.append(getScore(sum(temp)//len(temp)))
        else:
            avg.append(getScore((sum(temp)+myScore) // (len(temp)+1)))
    
    return "".join(avg)